src/litdata/utilities/shuffle.py

```diff
@@ -71,18 +71,39 @@ def _associate_chunks_and_intervals_to_workers(
     batch_size: int = 1,
 ) -> Tuple[List[List[int]], List[Any]]:
     num_items = sum([(interval[2] - interval[1]) for interval in chunk_intervals])
-    world_size = distributed_env.world_size * num_workers
-    num_items_per_workers: List[int] = [
-        num_items // world_size + num_items % world_size
-        if rank == world_size - 1 and not drop_last
-        else num_items // world_size
-        for rank in range(world_size)
-    ]
-    if drop_last:
-        num_items_per_workers = [batch_size * int(item // batch_size) for item in num_items_per_workers]
-
-    chunks_per_workers: List[List[int]] = [[] for _ in range(world_size)]
-    intervals_per_workers: List[List[List[int]]] = [[] for _ in range(world_size)]
+    max_batches = num_items // batch_size
+    global_num_workers = distributed_env.world_size * num_workers
+
+    num_items_per_workers: Any = []
+
+    for rank in range(distributed_env.world_size):
```

Comment:
why only dist worksize here and not global num workers?
same question for below.

Reply:
Because, we want to ensure we fill up all the workers for each process rank in the same way.